### PR TITLE
Add Windows acceptance tests

### PR DESCRIPTION
**Pull request by treydock**
# Pull Request Checklist

## Description

Add Windows acceptance tests
Support Windows 2016
Handle absence of os.architecture fact for Windows

## Related Issue

Migrated from https://github.com/sensu/sensu-puppet/pull/1052

### CODE REVIEW

**Comment by treydock** [comment]
Added appveyor to my namespace and it's failing because this module defaults to `gem` provider for sensu-plugin install and the rubygems version on appveyor is 3.0.2 which has no support for `--no-ri` and `--no-rdoc` that get executed by gem provider.

Defaulting all operating systems to `sensuclassic_gem` seems like the right move but is a breaking change as we'll be changing the default.  For now I'm updating the acceptance tests to change the default.

**Comment by treydock** [comment]
Using `sensuclassic_gem` seems to end up using system Ruby and causes issues with appveyor tests using Ruby 2.5.  Am trying to downgrade rubygems to 2.x in appveyor to avoid rubygems 3.x.

**Comment by ghoneycutt** [comment]
We don't want to change the code to support appveyor. Appveyor should allow us to run against the ruby versions we need.

**Comment by treydock** [comment]
We can but the pre-installed Ruby all have latest rubygems and latest rubygems breaks Puppet gem provider.  This wasn't an issue a few months ago when I did first appveyor tests but I think that's because rubygems 3.x wasn't released that drops support for --no-ri and --no-rdoc.  For now I'm just downgrading rubygems in appveyor installed Ruby.  Looks like Puppet hasn't been updated to support latest rubygems: https://github.com/puppetlabs/puppet/blob/master/lib/puppet/provider/package/gem.rb#L131

**Comment by treydock** [comment]
Successful build: https://ci.appveyor.com/project/treydock/puppet-module-sensuclassic/builds/23267606

File: appveyor.yml [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+- facter
+- ruby -v
+- gem update --system 2.7.9
+- gem -v
+- bundle -v
+- bundle install --jobs 4 --retry 2 --path C:\vendor\bundle
+test_script:
+- bundle exec rake acceptance:local
+image:
+  - Visual Studio 2015
```
Comment (ghoneycutt) [inline appveyor.yml:42]: which version(s) of windows does this test against?

**Comment by ghoneycutt** [comment]
I think I've got appveyor turned on now. Can you try pushing to a new branch or something to trigger it?

File: appveyor.yml (hunk 1 above)
Comment (treydock) [reply appveyor.yml:42]: https://www.appveyor.com/docs/windows-images-software/#operating-system

Visual Studio 2015 is 2012 R2 and Visual Studio 2017 is 2016.  I couldn't find a way to choose OS version except through Visual Studio.

Comment (ghoneycutt) [reply appveyor.yml:42]: Could you please add comments that document this so it is clear what we are testing against

**Comment by treydock** [comment]
Amended commit and forced push seems to have triggered appveyor.

File: appveyor.yml (hunk 1 above)
Comment (treydock) [reply appveyor.yml:42]: Comment added

**Comment by ghoneycutt** [comment]
@treydock could you please rebase this

**Comment by treydock** [comment]
Rebased

**Comment by ghoneycutt** [comment]
Released in v3.1.0  Thank you!

**Pull request merged**